src/main/kotlin/app/revanced/patches/youtube/layout/returnyoutubedislike/patch/ReturnYouTubeDislikePatch.kt

```diff
@@ -123,39 +130,44 @@ class ReturnYouTubeDislikePatch : BytecodePatch(
                             return PatchResultError("Method signature did not match: $this $parameterTypes")
 
                         val insertIndex = implementation!!.instructions.size - 1
-
                         val spannedParameterRegister = (instruction(insertIndex) as OneRegisterInstruction).registerA
                         val parameter = (instruction(insertIndex - 2) as BuilderInstruction35c).reference
 
                         if (!parameter.toString().endsWith("Landroid/text/Spanned;"))
                             return PatchResultError("Method signature parameter did not match: $parameter")
 
-                        addInstructions(
-                            insertIndex,
-                            """
-                                invoke-static {v$spannedParameterRegister}, $INTEGRATIONS_PATCH_CLASS_DESCRIPTOR->onShortsComponentCreated(Landroid/text/Spanned;)Landroid/text/Spanned;
-                                move-result-object v$spannedParameterRegister
-                            """
-                        )
+                        insertShorts(insertIndex, spannedParameterRegister)
                     }
                 }
+
+            // Additional hook, called after user dislikes.
+            with (it.mutableMethod) {
+                val insertIndex = it.scanResult.patternScanResult!!.startIndex + 2
+                val insertRegister = (implementation!!.instructions.elementAt(insertIndex - 1)
+                        as OneRegisterInstruction).registerA
+                insertShorts(insertIndex, insertRegister)
+            }
         } ?: return ShortsTextComponentParentFingerprint.toErrorResult()
 
-        // endregion
 
         return PatchResultSuccess()
     }
 
     private companion object {
         const val INTEGRATIONS_PATCH_CLASS_DESCRIPTOR =
-            "Lapp/revanced/integrations/patches/ReturnYouTubeDislikePatch;"
-
-        const val ON_COMPONENT_CREATED_DESCRIPTOR =
-            "$INTEGRATIONS_PATCH_CLASS_DESCRIPTOR->onComponentCreated(Ljava/lang/Object;Ljava/util/concurrent/atomic/AtomicReference;)V"
+            "Lapp/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike;"
 
         private fun MethodFingerprint.toPatch(voteKind: Vote) = VotePatch(this, voteKind)
     }
 
+    private fun MutableMethod.insertShorts(index: Int, register: Int) {
```

Comment:
Maybe accompany this extension method with MutableMethod.insertPlayer accordingly?